src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -313,7 +348,14 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
             // At least one full vector's worth of data remains, so we can safely read it.
             // Remember, at this point pBuffer is still aligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadAlignedVector128(pBuffer));
+            if (Sse2.IsSupported)
+            {
+                currentMask = (ulong)Sse2.MoveMask(Sse2.LoadAlignedVector128(pBuffer));
+            }
+            else
+            {
+                currentMask = GetMaskOfMostSignificantBits(AdvSimd.LoadVector128(pBuffer));
```

Comment:
Unless I'm completely misunderstanding something here, this implementation seems to assume that the two branches will compute the exact same value for `currentMask`. However this is not the case, for example if you take
```csharp
var vector = Vector128.Create((byte)128);
```
Then `Sse2.MoveMask(vector)` returns 0xFFFF, however `GetMaskOfMostSignificantBits` should return `0x1111_1111`.

Reply:
That's right. Since you need entire mask and not just the first lane set, you need longer implementation as suggested - https://github.com/dotnet/runtime/pull/39050#discussion_r454029220 and that should return `0xFFFF`.

Reply:
Ah you're right. I think I need to go back to using the original `AddPairwise` version you had